pkg/ebpf/state.go

```diff
@@ -247,22 +273,9 @@ func (ns *networkState) mergeConnections(id string, active map[string]*Connectio
 
 	// Active connections
 	for key, c := range active {
-		if closed, ok := client.closedConnections[key]; ok {
-			// If this connection was closed while we were collecting active connections it means the active
-			// connection is no more up-to date and we already went through the closed connection so let's
-			// skip it and not update the stats counters
-			if closed.LastUpdateEpoch >= c.LastUpdateEpoch {
-				continue
-			}
-
-			// If this connection was both closed and reopened, update the counters to reflect only the active connection.
-			// The monotonic counters will be the sum of all connections that cross our interval start + finish.
-			if stats, ok := client.stats[key]; ok {
-				stats.totalRetransmits = c.MonotonicRetransmits
-				stats.totalSent = c.MonotonicSentBytes
-				stats.totalRecv = c.MonotonicRecvBytes
-			}
-			continue // We processed this connection during the closed connection pass, so lets not do it again.
+		// If the connection was closed, it has already been processed so skip it
+		if _, ok := client.closedConnections[key]; ok {
+			continue
```

Comment:
we can't delete those from the `active` map because the `active` map is used after this function